### DISCUSSION TITLE
Add a table of contents to page that is on top of page rather than the right-hand page sidebar?

### DISCUSSION THREAD

**Discussion by cmdcolin** [facebook/docusaurus#3045] [General]
I am looking to have something similar to a wiki style TOC that is at the top of the page rather than the right-hand-sidebar for TOC

I am willing to dig into the code a little bit to see if I can make this possible if anyone has some pointers too

**Comment by cmdcolin** [comment]
Made a comment about this awhile ago to this effect here also https://v2.docusaurus.io/feedback/p/configuration-option-for-table-of-content

**Comment by slorber** [comment]
That should be possible to support, as a theme option like `docToc: "right" | "top"`

There's a feature called "rightToc" on the mdx parser, used in the theme in DocItem:

![image](https://user-images.githubusercontent.com/749374/87159248-547a4f80-c2c1-11ea-8f50-a1c506d92b8a.png)

You can probably use the same data `DocContent.rightToc` and render it at the top of a doc.
